libuserhelper.cpp

```diff
@@ -269,3 +300,33 @@ QString LibUserHelper::homeDir(uint uid)
 
     return rv;
 }
+
+QStringList LibUserHelper::groups(uint uid)
+{
+    struct lu_error *error = nullptr;
+    struct lu_context *context = lu_start(NULL, lu_user, NULL, NULL, NULL, NULL, &error);
+    if (!context) {
+        qCWarning(lcSUM) << "Error creating context:" << lu_strerror(error);
+        lu_error_free(&error);
+        return QStringList();
+    }
+
+    QStringList rv;
+    struct lu_ent *ent = lu_ent_new();
+    if (lu_user_lookup_id(context, uid, ent, &error)) {
+        const char *user = lu_ent_get_first_string(ent, LU_USERNAME);
+        GValueArray *groups = lu_groups_enumerate_by_user(context, user, &error);
+        if (groups) {
+            for (uint i = 0; i < groups->n_values; i++) {
+                rv.append(g_value_get_string(g_value_array_get_nth(groups, i)));
+            }
+        } else {
+            qCWarning(lcSUM) << "Error getting users groups:" << lu_strerror(error);
+            lu_error_free(&error);
```

Comment:
Side note: I took a peek at lu*error*() code. Looks "designed to be fragile" to me.

If error is set on entry to functions, library code will abort() - which is at least deterministic.

But then we have things like lu_groups_enumerate_by_user() which ensures error is not set when returning true, but AFAICT  *can* return false without setting error (when user==null) - in which case lu_strerror() here survives but lu_error_free() does null-deref crash and burn.

I have no idea how valid concern this might be.

Reply:
Added some error checks before freeing to avoid crashes. Indeed kibuser does not seem that robust.